### PR TITLE
Provide Github Release download source support for all platform

### PR DESCRIPTION
**Pull request by WillowSauceR**
Preview: 
preview

### CODE REVIEW

**Comment by WillowSauceR** [comment]
Preview screenshot of Ubuntu 22.04 platform(Powered by YsCloud docker):
OtherPlatformPreview

**Comment by Torrekie** [comment]
We provided an installation guide like this
```bash
bash -c "$(curl -fsSL https://storage.fastbuilder.pro/install.sh)"
```

In that scene, shell script was executed as literal, that means `argv` might not as expected as how we detect it oftentimes. Thus we choose to use environment variables to control some processes of this script (e.g. `LOCAL`)

**Comment by WillowSauceR** [comment]
> We provided an installation guide like this
> 
> ```shell
> bash -c "$(curl -fsSL https://storage.fastbuilder.pro/install.sh)"
> ```
> 
> In that scene, shell script was executed as literal, that means `argv` might not as expected as how we detect it oftentimes. Thus we choose to use environment variables to control some processes of this script (e.g. `LOCAL`)
Corresponding revisions have been completed:
屏幕截图_20221111_221707

**Review by Torrekie — CHANGES_REQUESTED** [review]

File: install.sh [hunk 1]
```diff
@@ -350,10 +357,27 @@ report_error() {
...
+  printf "Downloading FastBuilder binary...\n"
   # Repeat FB_LINK
   FB_LINK="${FB_DOMAIN}${FB_LOCATION_ROOT}${FB_PREFIX}${FILE_ARCH}${FILE_TYPE}${FILE_ARCH}"
-  printf "Downloading FastBuilder binary..."
+  if [[ ${PB_USE_GH_REPO} == "1" ]]; then
+    printf "\033[32mOriginal download link: ${FB_LINK}\033[0m\n"
+    FB_LINK="${GH_LINK}v${FB_VER}/${FB_PREFIX}${FILE_ARCH}${FILE_TYPE}${FILE_ARCH}"
+    printf "\033[32mGithub download link: ${FB_LINK}\033[0m\n"
+  fi
+  printf "\033[33mIf the download is too slow, you can press \"Ctrl+C\" to end the current script and later try to use \"export PB_USE_GH_REPO=1\" and then re-run the installation command to switch to the Github Releases source to download the PhoenixBuilder package.\033[0m\n"
```
Comment (LNSSPsd) [inline install.sh:380]:
```suggestion
  printf "\033[33mIf the official storage does not work for you, you can try to assign environment variable \"PB_USE_GH_REPO=1\" for the script to download stuff from Github.\033[0m\n"
```

File: install.sh [hunk 2]
```diff
@@ -378,37 +402,15 @@ if [[ ${BINARY_INSTALL} == "1" ]]; then
...
-  fi
   printf "Downloading FastBuilder package...\n"
   # Repeat FB_LINK
   FB_LINK="${FB_DOMAIN}${FB_LOCATION_ROOT}${FB_PREFIX}_${FB_VER}_${FILE_ARCH}${FILE_TYPE}"
+  if [[ ${PB_USE_GH_REPO} == "1" ]]; then
+    printf "\033[32mOriginal download link: ${FB_LINK}\033[0m\n"
+    FB_LINK="${GH_LINK}v${FB_VER}/${FB_PREFIX}_${FB_VER}_${FILE_ARCH}${FILE_TYPE}"
+    printf "\033[32mGithub download link: ${FB_LINK}\033[0m\n"
+  fi
+  printf "\033[33mIf the download is too slow, you can press \"Ctrl+C\" to end the current script and later try to use \"export PB_USE_GH_REPO=1\" and then re-run the installation command to switch to the Github Releases source to download the PhoenixBuilder package.\033[0m\n"
```
Comment (LNSSPsd) [inline install.sh:413]: See the suggested change above.

**Review by LNSSPsd — COMMENTED** [review]
Just a suggestion, changes not requested, it's not mandatory to do like that.

**Review by Torrekie — APPROVED** [review]

**Review by LNSSPsd — APPROVED** [review]

**Pull request merged**